### DISCUSSION TITLE
prise en charge version EP

### DISCUSSION THREAD

**Discussion by jeremGi** [cmcgerty/Sofar2mqtt#57] [General]
Bonjour

En permier merci pour le travail de developpement du sofar2mqtt.

je possede un HYD5000EP donc les codes pour le registre n'etaient pas bon, j'ai trouvé via un autre projet les adresses pour les infos que je integré dans ma version de sofar2mqtt.
les infos remontent bien mais j'ai cependant des problemes de CRC et des plantages du wemos qui le font redemarré.

ci joint le code que j'utilise si jamais vous voulez faire une version qui prenne en charge tous les modeles donc les HYD xxxx EP
je vous envoi aussi le log du moniteur serie avec le plantage en question.

[log.txt](https://github.com/cmcgerty/Sofar2mqtt/files/10831814/log.txt)
[Sofar2mqtt_HYD5000EP_test_github.zip](https://github.com/cmcgerty/Sofar2mqtt/files/10831817/Sofar2mqtt_HYD5000EP_test_github.zip)

**Comment by rjpearce** [comment]
193 `#define SOFAR_REG_GRIDA   0x0207 //codes non trouvés`
194  `#define SOFAR_REG_GRIDW   0x0212 //codes non trouvés`

These could be the source of the CRC errors as they aren't valid register addresses on the EP.

`//codes non verifies
#define SOFAR_FN_STANDBY  0x0100 
#define SOFAR_FN_DISCHARGE  0x0101
#define SOFAR_FN_CHARGE   0x0102
#define SOFAR_FN_AUTO   0x0103`

For these checkout the sample code in this Facebook post:
https://www.facebook.com/groups/2477195449252168/permalink/3047934172178290/

You could also put the different values in a HYD-EP.h file and then include it #include "HYD-EP.h", it might make it easier to maintain.

**Reply by joostvdveen** [reply]
Hi @jeremGi i am trying to get my HYD 5000-EP up and running, started with the standard sofar2mqtt but that dit not work out since it is for the ES and then came across your post with the right addresses for the EP.
Question, it looks like the values in Home Assistant are not correct, how dit you deal with that, can you share your config syntax of configuration.yaml ?

**Reply by jeremGi** [reply]
bonjour
désolé je suis sous jeedom et non home assistant.
les valeurs incorrectes sont celles de la batterie en décharge ?

**Comment by joostvdveen** [comment]
Mine crashes too after a couple of minutes...it does not impact anything as fas as I know, just crashes and restarts

```c++
3fffff30:  00000587 3ffeed14 3ffeec74 4020276c  
3fffff40:  616c6f53 32767072 09000076 3ffef030  
3fffff50:  3fffdad0 00001388 3ffefed4 00f800ff  
3fffff60:  81000000 3ffeeda4 3ffe897c 4020252a  
3fffff70:  3fffdad0 3ffeeda4 3ffe897c 4020289f  
3fffff80:  00000000 0004d000 00000000 00000000  
3fffff90:  69746300 00000000 00000000 3f00656e  
3fffffa0:  00000000 00000000 3ffef000 00000000  
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------
```

**Reply by jeremGi** [reply]
oui j'ai arrete de trouver une solution ca marche comme ca
